Lib/idlelib/autocomplete_w.py

```diff
@@ -153,6 +173,39 @@ def _selection_changed(self):
                 self.listbox.select_set(self._binary_search(self.start))
                 self._selection_changed()
 
+    @classmethod
+    def _quote_closes_literal(cls, start, quotechar):
+        """Check whether the typed quote character closes a str/bytes literal.
+
+        Note that start doesn't include the typed character.
+        """
+        quote_index = _find_first_quote(start)
+        if not (quote_index >= 0 and start[quote_index] == quotechar):
+            return False
+
+        # Recognize triple-quote literals.
+        if start[quote_index:quote_index + 3] == quotechar * 3:
+            # The start string must end with two such quotes.
+            if start[-2:] != quotechar * 2:
+                return False
+            quote = quotechar * 3
+        else:
+            quote = quotechar
+
+        # The start string must be long enough.
+        if len(start) - quote_index < len(quote) * 2 - 1:
+            return False
+
+        # If there was an odd number of back-slashes just before
+        # the closing quote(s), then the quote hasn't been closed.
+        i = len(start) - (len(quote) - 1)
+        while (
+                i >= quote_index + len(quote) + 2 and
+                start[i - 2:i] == '\\\\'
+        ):
+            i -= 2
+        return start[i - 1] != '\\'
+
     def show_window(self, comp_lists, index, complete, mode, userWantsWin):
```

Comment:
Could we potentially change the parameter `userWantsWin` -> `user_wants_win` or `wants_win`? This already existed prior to this change, but it looks a bit odd to have one parameter within the function use the typical underscore naming convention and another using camelCase.

```suggestion
    def show_window(self, comp_lists, index, complete, mode, wants_win):
```

Reply:
Yes, we have be gradually PEP8-ifying idlelib as modules are otherwise reviewed and edited *and tests added*.  And shorter is fine here.

Reply:
This argument is actually no longer used for anything as far as I can tell. I double-check and then remove it.

Reply:
Removed in latest commit.